src/node.cc

```diff
@@ -4956,6 +4956,15 @@ int Start(int argc, char** argv) {
 #if ENABLE_TTD_NODE
   bool chk_debug_enabled = debug_options.inspector_enabled();
 
+  std::string envDoRecordVar;
+  bool envDoRecord = SafeGetenv("DO_TTD_RECORD", &envDoRecordVar) &&
```

Comment:
Each process gets a copy of the parent's environment variables, so any changes are localized to that process.